### PR TITLE
ORC-1486: Fixes checkstyle violation in test classes for `orc-core` module

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR fixes checkstyle violation in test sources in orc-core module.

`Indentation` rule is suppressed in some files explicitly.

### Why are the changes needed?

The change shall ensure our test code adhere's to same formatting as our source

### How was this patch tested?

Remove the following and run Checkstyle on `core` module.

https://github.com/apache/orc/blob/7787669c444d0cf18ba91effbba34b5608370b5b/java/checkstyle-suppressions.xml#L31-L36

```
$ cd java

$ git diff
diff --git a/java/checkstyle-suppressions.xml b/java/checkstyle-suppressions.xml
index 32e04a588..0256e3b77 100644
--- a/java/checkstyle-suppressions.xml
+++ b/java/checkstyle-suppressions.xml
@@ -44,10 +44,4 @@
   <suppress checks="CustomImportOrder" files="src/test/*"/>

   <!-- Remove the following rule when the test code clean up is completed. -->
-  <suppress checks="Indentation" files="src/test/*"/>
-  <suppress checks="RedundantModifier" files="src/test/*"/>
-  <suppress checks="ModifierOrder" files="src/test/*"/>
-  <suppress checks="UpperEll" files="src/test/*"/>
-  <suppress checks="NeedBraces" files="src/test/*"/>
-  <suppress checks="RegexpSingleline" files="src/test/*"/>
 </suppressions>

$ mvn checkstyle:check --pl core
Running `/Users/dongjoon/APACHE/orc-merge/java/mvnw`...
Using `mvn` from path: /opt/homebrew/bin/mvn
[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------< org.apache.orc:orc-core >-----------------------
[INFO] Building ORC Core 2.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- checkstyle:3.3.0:check (default-cli) @ orc-core ---
[INFO] You have 0 Checkstyle violations.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.228 s
[INFO] Finished at: 2023-08-25T22:13:02-07:00
[INFO] ------------------------------------------------------------------------
```

This closes #1590

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @mystic-lama

**Review by dongjoon-hyun — COMMENTED** [review]
I hope we can proceed in this way module by module, @mystic-lama .

**Comment by mystic-lama** [comment]
@dongjoon-hyun Thank you so much. I only get time during weekend or evening to work on OSS. Sadly got pulled into work, which took that time. I shall try to close things faster. Let me catch up with things and we can get this done ASAP

File: java/core/src/test/org/apache/orc/TestNewIntegerEncoding.java
Comment (mystic-lama) [inline java/core/src/test/org/apache/orc/TestNewIntegerEncoding.java:340]: Shall we have this block like this?  I find it more easy on my eyes. We don't have to fix these things now. I can take care of this later

```
long[] inp = new long[]{4513343538618202719L, 4513343538618202711L,
                        2911390882471569739L, -9181829309989854913L};                                                                               
```

**Review by mystic-lama — APPROVED** [review]
+1 LGTM

Thanks a lot @dongjoon-hyun

File: java/core/src/test/org/apache/orc/TestNewIntegerEncoding.java
Comment (dongjoon-hyun) [reply java/core/src/test/org/apache/orc/TestNewIntegerEncoding.java:340]: May I asked which rule complains about that, @mystic-lama ?

As you see, this PR aims to introduce the minimal changes which means to touch only `Checkstyle` really complains. The verification is described in the PR description, too.

To simply put, if we mix different themes arbitrarily like that, it tends to make a mistake because of the human error factor. It makes reviewers difficulty to review the PRs. It ends up as a situation we cannot merge it.

Of course, we can take care of it later independently.

**Comment by dongjoon-hyun** [comment]
BTW, please note that Apache ORC has 3 year maintenance policy. It means ORC committers need to maintain `branch-1.9` and `branch-1.8` and `branch-1.7` for three years and usually it requires us to backport from main to `branch-1.9` to `branch-1.8` and `branch-1.7` sequentially if needed. So, we are unable to accept a drastic change like your previous PR even for one time activity. It will tease us for 3 years.

**Comment by dongjoon-hyun** [comment]
Merged to main for Apache ORC 2.0.

```
$ git log -n1 | tail -n2
    Authored-by: mystic-lama <mysticlama000@gmail.com>
    Signed-off-by: Dongjoon Hyun <dongjoon@apache.org>
```

**Comment by dongjoon-hyun** [comment]
Let me make a PR for the rest of module. It seems that we can close this checkstyle task today.

**Comment by dongjoon-hyun** [comment]
Here is the PR.
#1601

**Comment by mystic-lama** [comment]
> BTW, please note that Apache ORC has 3 year maintenance policy. It means ORC committers need to maintain `branch-1.9` and `branch-1.8` and `branch-1.7` for three years and usually it requires us to backport from main to `branch-1.9` to `branch-1.8` and `branch-1.7` sequentially if needed. So, we are unable to accept a drastic change like your previous PR even for one time activity. It will tease us for 3 years.

@dongjoon-hyun This is good to know.

File: java/core/src/test/org/apache/orc/TestNewIntegerEncoding.java
Comment (mystic-lama) [reply java/core/src/test/org/apache/orc/TestNewIntegerEncoding.java:340]: I agree with your point. I don't recall the rule, and did not mean to fix it in this PR
This is more for future, when we make code changes. I shall keep an eye for this and next time it hits, may be we can refine the rule.

**Pull request closed without merging**
